src/operations/query/r4.js

```diff
@@ -17,6 +17,129 @@ const {searchParameterQueries} = require('../../searchParameters/searchParameter
+    if (queryParameterValue) {
+        // un-bundle any objects coming from graphql
+        if (typeof queryParameterValue === 'object' && !Array.isArray(queryParameterValue) && queryParameterValue['searchType']) {
+            switch (queryParameterValue['searchType']) {
+                case 'string':
+                    // handle SearchString
+                    if (queryParameterValue['value']) {
+                        queryParameterValue = queryParameterValue['value'];
+                    } else if (queryParameterValue['values']) {
+                        queryParameterValue = queryParameterValue['values'];
+                    }
+                    break;
+                case 'token':
+                    if (queryParameterValue['value']) {
+                        queryParameterValue['values'] = [queryParameterValue['value']];
+                    }
+                    if (queryParameterValue['values']) {
+                        for (const token of queryParameterValue['values']) {
+                            queryParameterValue = [];
+                            let tokenString = '';
+                            if (token['system']) {
+                                tokenString = token['system'] + '|';
+                            }
+                            if (token['code']) {
+                                tokenString += token['code'];
+                            }
+                            if (token['value']) {
+                                tokenString += token['value'];
+                            }
+                            if (tokenString) {
+                                queryParameterValue.push(tokenString);
+                            }
+                        }
+                    }
+                    break;
+                case 'reference':
+                    // eslint-disable-next-line no-case-declarations
+                    let referenceText = '';
+                    if (queryParameterValue['target']) {
+                        referenceText = queryParameterValue['target'] + '/';
+                    }
+                    if (queryParameterValue['value']) {
+                        referenceText += queryParameterValue['value'];
+                    }
+                    queryParameterValue = referenceText;
+                    break;
+                case 'quantity':
+                    // eslint-disable-next-line no-case-declarations
+                    let quantityString = '';
+                    if (queryParameterValue['prefix']) {
+                        quantityString += queryParameterValue['prefix'];
+                    }
+                    if (queryParameterValue['value']) {
+                        quantityString += queryParameterValue['value'];
+                    }
+                    if (queryParameterValue['system']) {
+                        quantityString = '|' + queryParameterValue['system'];
```

Comment:
Would this overwrite anything set from the above two 'if' statements if they were true?

Reply:
that's the idea.  it is a concatenation if multiple are specified.